src/Entity/User.php

```diff
@@ -83,7 +83,7 @@ public function __construct(string $login, string $password)
 
     public function getId(): ?string
     {
-        return (string)$this->id;
+        return $this->id === null ? null : (string)$this->id;
```

Comment:
Why casting is needed? Why didn't you make returned type as `?int`?

Reply:
https://github.com/yiisoft/auth/blob/93d5e85740de27a6e93941a4cae8fc97d37ffc7d/src/IdentityInterface.php#L11